cppcache/src/SerializationRegistry.cpp

```diff
@@ -151,31 +151,29 @@ std::shared_ptr<Serializable> SerializationRegistry::deserialize(
       return enumObject;
     }
     case DSCode::CacheableUserData: {
-      compId |= ((static_cast<int64_t>(input.read())) << 32);
+      classId = input.read();
       break;
     }
     case DSCode::CacheableUserData2: {
-      compId |= ((static_cast<int64_t>(input.readInt16())) << 32);
+      classId = input.readInt16();
       break;
     }
     case DSCode::CacheableUserData4: {
-      int32_t classId = input.readInt32();
-      compId |= ((static_cast<int64_t>(classId)) << 32);
+      classId = input.readInt32();
       break;
     }
     case DSCode::FixedIDByte: {
-      compId = input.read();
+      classId = input.read();
```

Comment:
This is a fixedId and probably shouldn't be confused with a classId. Maybe when you extract the handler functions it will be more clear.